docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,491 @@
+Graphical representation of exercises
```

Comment:
This part will age very fast. OpenAI already released GPT4 with support for pictures. Other similar tools also already support images. This image support will only improve in the future and eventually such multi-modal LLMs will handle graphical based problem statements just as well as textual ones.

Also for UML diagrams this would mean that we could no longer use PlantUML diagram for easy highlighting of what is correct/wrong in the current submission, as those are given in the problem statement as text and can be easily understood by an LLM.

Another thing to consider here is that this would also make Artemis a lot less accessible (more than it already is with UML diagrams) for students with visual impairments, who might rely on screenreaders or other tools. Might not be a huge issue, but still not great.

Reply:
I certainly can understand the point about inaccessability for Artemis. We can not prevent the use of LLMs by students in general. The recommendations are more to disencourage students to use this tools, since it requires little more effort. There are still ways to evade all the recommendations. GPT4 is just for paid users and copying UMLs etc is more effort than just copying the problem statement straight to these tools. I will adapt it so that we do not collide with inaccessability with the User Interface.